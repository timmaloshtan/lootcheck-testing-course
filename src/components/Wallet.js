import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Wallet extends Component {
  render() {
    const { balance } = this.props;
    return (
      <div className="wallet">
        <h3 className="wallet__balance">Wallet balance: {balance}</h3>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  balance: state,
});

export default connect(mapStateToProps, null)(Wallet);
