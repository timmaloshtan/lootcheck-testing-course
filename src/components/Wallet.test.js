import React from 'react';
import { shallow } from 'enzyme';

import { Wallet } from './Wallet';

const props = {
  balance: 0,
};

describe('Wallet component', () => {
  const wallet = shallow(<Wallet {...props} />);

  it('should render correctly', () => {
    expect(wallet).toMatchSnapshot();
  });

  it('should display the balance from props', () => {
    expect(wallet.find('.wallet__balance').text()).toEqual('Wallet balance: 0');
  });
  
});
