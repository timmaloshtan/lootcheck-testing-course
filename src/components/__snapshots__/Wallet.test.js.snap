// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet component should render correctly 1`] = `
<div
  className="wallet"
>
  <h3
    className="wallet__balance"
  >
    Wallet balance: 
    0
  </h3>
</div>
`;
